include/seqan3/io/sam_file/format_bam.hpp

```diff
@@ -961,6 +945,48 @@ inline void format_bam::write_alignment_record([[maybe_unused]] stream_type & st
     } // if constexpr (!detail::decays_to_ignore_v<header_type>)
 }
 
+//!\copydoc seqan3::detail::format_sam_base::write_header
+template <typename stream_t, typename header_type>
+inline void format_bam::write_header(stream_t & stream, sam_file_output_options const & options, header_type & header)
+{
+    if constexpr (detail::decays_to_ignore_v<header_type>)
+    {
+        throw format_error{"BAM can only be written with a header but you did not provide enough information! "
+                           "You can either construct the output file with reference names and reference length "
+                           "information and the header will be created for you, or you can access the `header` member "
+                           "directly."};
+    }
+    else
+    {
+        detail::fast_ostreambuf_iterator stream_it{*stream.rdbuf()};
+
+        std::ranges::copy_n("BAM\1", 4, stream_it);
+
+        // write SAM header to temporary stream first to query its size.
+        std::ostringstream os;
+        detail::format_sam_base::write_header(os, options, header);
+        int32_t l_text{static_cast<int32_t>(os.str().size())};
```

Comment:
```suggestion
        int32_t const l_text{static_cast<int32_t>(os.view().size())};
```

`std::ostringstream::view()` is CPP20, let's hope, GCC 10 implements it